### PR TITLE
TGU: Agent ID buff, Even more Circuit stuff, Spectroscopy, DBcore subsystem to help with crashes

### PR DESCRIPTION
**Pull request by SgtHunk**
## About The Pull Request

-  Okay, so. First big thing here: DBcore now works off a subsystem if executed async, and limits the maximum amount of concurrent async queries to 25. This adds "MAX_CONCURRENT_QUERIES" to the dbconfig, so don't forget to add it. Yes this helps with crashes.
- Agent ID cards can now jump to the front of wallets.
- Contractor Baton's cyborg stun now uses left click to stun/right click to harmbaton.
- Mutant bodyparts/mutcolors are now editable genetic traits.
- Robotics now has an ectostopic sniffer. Ghosts can click on it to make a 'ping'. You can disable it by cutting the single wire in it.
- USB cables for circuits (they connect to computers/machines).
- Juke Build.

And a lot more I probably forgot to mention but it's mainly bugfixes so
this is just for the common reader go look at tg's merged PRs if you really want to know

TODO:

- [ ] Turn all of our code's spans into the macros.

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -101,10 +101,6 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 		cmd_admin_pm(href_list["priv_msg"],null)
 		return
 
-	// Fulpstation Mentorhelp
-	if(mentor_client_procs(href_list))
```
Comment (JohnFulpWillard) [inline code/modules/client/client_procs.dm:105]: fulp edit

**Review by JohnFulpWillard — COMMENTED** [review]
im gonna cry

File: code/modules/mob/living/carbon/human/examine.dm [hunk 2]
```diff
@@ -16,15 +14,15 @@
 		if(HAS_TRAIT(L, TRAIT_PROSOPAGNOSIA))
 			obscure_name = TRUE
 
+	. = list("<span class='info'>*---------*\nThis is <EM>[!obscure_name ? name : "Unknown"]</EM>!")
```
[the remark this answers is not in the document]
Comment (SgtHunk) [reply code/modules/mob/living/carbon/human/examine.dm]: Trust me, I also have no idea
I pulled off of master and for some reason it did that

File: code/modules/mob/living/carbon/human/examine.dm [hunk 3]
```diff
@@ -196,39 +196,41 @@
...
 			if(temp < 25)
-//				msg += "[t_He] [t_has] minor bruising.\n"
-				msg += "[t_He] [t_has] minor [dna.species.bruising_desc].\n" // FULP EDIT START - Species Descriptors
+				// msg += "[t_He] [t_has] minor bruising.\n"
+				msg += "[t_He] [t_has] minor [dna.species.bruising_desc].\n"
 			else if(temp < 50)
-//				msg += "[t_He] [t_has] <b>moderate</b> bruising!\n"
-				msg += "[t_He] [t_has] <b>moderate</b> [dna.species.bruising_desc]!\n"
+				// msg += "[t_He] [t_has] <b>moderate</b>  bruising!\n"
+				msg += "[t_He] [t_has] <b>moderate</b>  [dna.species.bruising_desc]!\n"
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/human/examine.dm:205]: You've added an extra space here.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 4]
```diff
@@ -196,39 +196,41 @@
...
 			else
-//				msg += "<B>[t_He] [t_has] severe bruising!</B>\n"
-				msg += "<B>[t_He] [t_has] severe [dna.species.bruising_desc]!</B>\n"
+				// msg += "<B>[t_He] [t_has] severe  bruising!</B>\n"
+				msg += "<B>[t_He] [t_has] severe  [dna.species.bruising_desc]!</B>\n"
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/human/examine.dm:208]:
```suggestion
				// msg += "<B>[t_He] [t_has] severe bruising!</B>\n"
				msg += "<B>[t_He] [t_has] severe [dna.species.bruising_desc]!</B>\n"
```

File: code/modules/mob/living/carbon/human/examine.dm (hunk 3 above)
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/human/examine.dm:205]:
```suggestion
				// msg += "[t_He] [t_has] <b>moderate</b> bruising!\n"
				msg += "[t_He] [t_has] <b>moderate</b> [dna.species.bruising_desc]!\n"
```

**Pull request merged**